src/bits.rs

```diff
@@ -305,12 +305,39 @@ impl<const BITS: usize, const LIMBS: usize> Uint<BITS, LIMBS> {
     /// the shift is larger than `BITS` (which is IMHO not very useful).
     #[inline]
     #[must_use]
-    pub const fn overflowing_shl(self, rhs: usize) -> (Self, bool) {
+    pub const fn overflowing_shl(mut self, rhs: usize) -> (Self, bool) {
+        if rhs == 0 {
+            // if there are no limbs and no bits, return self
+            return (self.masked(), false);
```

Comment:
no need to mask an already valid uint; only when you modify the raw limbs is this needed